yrs/src/doc.rs

```diff
@@ -209,13 +293,75 @@ pub enum OffsetKind {
     Utf32,
 }
 
+pub trait Transact {
+    /// Creates a transaction used for all kind of block store operations.
+    /// Transaction cleanups & calling event handles happen when the transaction struct is dropped.
+    fn try_transact(&self) -> Result<Transaction, BorrowError>;
+
+    /// Creates a transaction used for all kind of block store operations.
+    /// Transaction cleanups & calling event handles happen when the transaction struct is dropped.
+    fn try_transact_mut(&self) -> Result<TransactionMut, BorrowMutError>;
+
+    /// Creates a transaction used for all kind of block store operations.
+    /// Transaction cleanups & calling event handles happen when the transaction struct is dropped.
+    fn transact(&self) -> Transaction {
+        self.try_transact().expect("cannot read document store contents because another read-write transaction is in progress")
+    }
+
+    /// Creates a transaction used for all kind of block store operations.
+    /// Transaction cleanups & calling event handles happen when the transaction struct is dropped.
+    fn transact_mut(&self) -> TransactionMut {
```

Comment:
From the docs the difference between `transact` and `transact_mut` is not clear.

The `*_mut` part might be confusing, because both transaction types return an object. Something like `transact_rw` & `transact_ro` might be better.

Reply:
This is something that I've seen in other rust libraries. `mut` suffix is a standard for methods returning objects that can be mutated or used to mutate data structures (eg. `RefCell::borrow` returns `Ref` which is readonly reference, while `RefCell::borrow_mut` returns `RefMut` which is mutable reference). For this reason I decided to keep that naming convention in yrs.

Reply:
Seconded that _mut is understood in the rust ecosystem.